pkg/streamingpromql/operators/functions/times.go

```diff
@@ -62,7 +54,6 @@ func timeWrapperFunc(f func(t time.Time) float64) InstantVectorSeriesFunction {
 		}
 
 		// we don't do time based function on histograms
-		types.HPointSlicePool.Put(seriesData.Histograms, memoryConsumptionTracker)
 		seriesData.Histograms = nil
```

Comment:
This needs to be done before we return `seriesData` above as well.

In fact, we could now drop the `if len(seriesData.Floats) > 0` check entirely, which would remove the early return as well.

Reply:
I moved this up now, but not inside defer block. Because putting them in defer block causing the pedantic memory pool test to fail complaining negative allocation.